mockwebserver/src/main/kotlin/mockwebserver3/MockWebServer.kt

```diff
@@ -337,26 +317,46 @@ public class MockWebServer : Closeable {
   /**
    * Starts the server and binds to the given socket address.
    *
-   * @param inetSocketAddress the socket address to bind the server on
+   * @param socketAddress the socket address to bind the server on
    */
   @Synchronized
   @Throws(IOException::class)
-  private fun start(inetSocketAddress: InetSocketAddress) {
+  private fun start(socketAddress: InetSocketAddress) {
     check(!closed) { "shutdown() already called" }
-    if (started) return
-    started = true
 
-    this._inetSocketAddress = inetSocketAddress
+    val alreadyStartedAddress = socketAddress_
+    if (alreadyStartedAddress != null) {
```

Comment:
Start used to silently ignore calls after it was started, even if the target address had changed